### PR TITLE
[batch-rule] householder_product

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: https://github.com/pytorch/functorch/issues/240

### CODE REVIEW

File: functorch/csrc/BatchRulesLinearAlgebra.cpp [hunk 1]
```diff
@@ -160,6 +178,7 @@ TORCH_LIBRARY_IMPL(aten, FT_BATCHED_KEY, m) {
   VMAP_SUPPORT("mv", mv_batch_rule);
   VMAP_SUPPORT("mm", mm_batch_rule);
   m.impl("linear", linear_decomp);
+  m.impl("orgqr", orgqr_decomp);
```
Comment (kshitij12345) [inline functorch/csrc/BatchRulesLinearAlgebra.cpp:181]: Without this we get 
`RuntimeError: aten::orgqr hit the vmap fallback which is currently disabled`

But `orgqr` is a composite operator 🤔.

Comment (vfdev-5) [reply functorch/csrc/BatchRulesLinearAlgebra.cpp:181]: Maybe, you can add it into BatchRulesDecompositions.cpp

File: functorch/csrc/BatchRulesLinearAlgebra.cpp [hunk 2]
```diff
@@ -151,6 +151,20 @@ Tensor linear_decomp(
   return result;
 }
 
+std::tuple<Tensor, c10::optional<int64_t>>
```
Comment (Chillee) [inline functorch/csrc/BatchRulesLinearAlgebra.cpp:154]: Can we use `EXISTING_BDIM` or `EXISTING_BDIM_ALL_BOXED` for this?

Comment (kshitij12345) [reply functorch/csrc/BatchRulesLinearAlgebra.cpp:154]: Unfortunately neither of them works. 
* `EXISTING_BDIM` seems to assume that only `self` has bdim 
* With `EXISTING_BDIM_ALL_BOXED` we get `torch.linalg.householder_product: input.shape[-1] must be greater than or equal to tau.shape[-1]`

**Comment by zou3519** [comment]
@Chillee @kshitij12345 can we merge this?

**Comment by Chillee** [comment]
@zou3519 I'm fine with merging it.

**Comment by zou3519** [comment]
Merged in https://github.com/pytorch/functorch/pull/972

**Pull request closed without merging**